docs/development.rst

```diff
@@ -26,157 +121,606 @@ Quickstart
+checks for code quality.
+
+- A `Github action <https://help.github.com/en/actions>`_ runs some code style and quality checks.
+
+- `CircleCI <https://circleci.com/>`_ runs all Python tests using the apt-get
+  infrastructure for system requirements. We also runs C tests, compiled
+  using gcc and clang, and check for memory leaks using valgrind.
+  Documentation is also built to check for any errors.
+
+- `Travis CI <https://travis-ci.org/>`_ runs Python tests on Linux and OSX using the
+  `Conda <https://conda.io/docs/>`__ infrastructure for the system level
+  requirements.
+
+- `AppVeyor <https://www.appveyor.com/>`_ runs Python tests on
+  Windows using conda.
+
+- `CodeCov <https://codecov.io/gh>`__ tracks test coverage in Python and C.
+
+
+.. _sec_development_best_practices:
 
 ******************************
 Best Practices for Development
 ******************************
 
 The following is a rough guide of best practices for contributing a function to the
 tskit codebase.
-We will use the :meth:`Tree.kc_distance` method as an example. 
+We will use the :meth:`Tree.kc_distance` method as an example.
+
+Note that this guide covers the most complex case of adding a new function to both
+the C and Python APIs.
 
 0.  Open an `issue <http://github.com/tskit-dev/tskit/issues>`_ with your proposed
     functionality. If consensus is reached that your proposed addition should be
-    added to the codebase, proceed! 
-1.  Create a new branch on your fork of tskit-dev (see :ref:`sec_development_quickstart`
+    added to the codebase, proceed!
+1.  Create a new branch on your fork of tskit-dev (see :ref:`sec_development_getting_started`
     above). Then open a `pull request <http://github.com/tskit-dev/tskit/pulls>`_ on GitHub,
     with an initial description of your planned addition.
-2.  Write your function in Python: in python/tests find the test module that 
+2.  Write your function in Python: in ``python/tests/`` find the test module that
     pertains to the functionality you wish to add. For instance, the kc_distance
-    metric was added to 
+    metric was added to
     `test_topology.py <https://github.com/tskit-dev/tskit/blob/master/python/tests/test_topology.py>`_.
     Add a python version of your function here.
 3.  Create a new class in this module to write unit tests for your function: in addition
     to making sure that your function is correct, make sure it fails on inappropriate inputs.
-    This can often require judgement. For instance, :meth:`Tree.kc_distance` fails on a tree 
+    This can often require judgement. For instance, :meth:`Tree.kc_distance` fails on a tree
     with multiple roots, but allows users to input parameter values that are nonsensical,
-    as long as they don't break functionality. 
+    as long as they don't break functionality. See the
+    `TestKCMetric <https://github.com/tskit-dev/tskit/blob/4e707ea04adca256036669cd852656a08ec45590/python/tests/test_topology.py#L293>` for example.
```

Comment:
link not rendering - missing trailing underscore?

Reply:
```suggestion
    `TestKCMetric <https://github.com/tskit-dev/tskit/blob/4e707ea04adca256036669cd852656a08ec45590/python/tests/test_topology.py#L293>`_ for example.
```